hr_leave_has_attachment/models/ir_attachment.py

```diff
@@ -0,0 +1,25 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import models, api
+
+
+class IrAttachment(models.Model):
+    _inherit = 'ir.attachment'
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super(IrAttachment, self).create(vals_list)
+        for record in res.filtered(lambda x: x.res_model == 'hr.leave'):
+            hr_leave_id = self.env['hr.leave'].browse(record.res_id).exists()
+            hr_leave_id.has_attachment = True
```

Comment:
Try this "recordset based" code (not tested), for better optimization (only one `browse` operation). And I think that `exists()` reference is unnecessary (when an attachment is added, related leave it's supposed to exist every time):
```suggestion
        leave_id_list = res.filtered(lambda x: x.res_model == 'hr.leave').mapped("res_id")
        if leave_ids:
            leave_ids = self.env["hr.leave"].browse(leave_id_list)
            leave_ids.filtered(lambda x: not x.has_attachment).write({"has_attachment": True})
```